static/app/views/projectInstall/issueAlertOptions.tsx

```diff
@@ -314,6 +320,24 @@ class IssueAlertOptions extends DeprecatedAsyncComponent<Props, State> {
       this.state.conditions?.length > 0
     );
 
+    // Hack to clear out the alert setting if the platform changes to python or
+    // javascript, which should have a different default alert type
+    if (
+      this.shouldUseNewDefaultSetting() &&
+      this.state.alertSetting === RuleAction.ALERT_ON_EVERY_ISSUE.toString()
+    ) {
+      this.setStateAndUpdateParents({
+        alertSetting: RuleAction.ALERT_ON_HIGH_PRIORITY_ISSUES.toString(),
+      });
+    } else if (
+      !this.shouldUseNewDefaultSetting() &&
+      this.state.alertSetting === RuleAction.ALERT_ON_HIGH_PRIORITY_ISSUES.toString()
+    ) {
+      this.setStateAndUpdateParents({
+        alertSetting: RuleAction.ALERT_ON_EVERY_ISSUE.toString(),
+      });
+    }
```

Comment:
can't set state inside render funcitons. this would need to be in a lifecycle event like componentDidUpdate